### PR TITLE
error handling

### PR DESCRIPTION
**Pull request by lakshith-403**

### CODE REVIEW

File: app/ui/src/network.ts [hunk 1]
```diff
@@ -216,8 +216,8 @@ class Network {
                 }
             }
 
-            xhr.onerror = () => {
-                reject('Network Failure')
+            xhr.onerror = (event) => {
+                reject(new Error(`XHR request failed: ${event}\n Type: ${event.type}: ${event.loaded} bytes transferred`))
```
Comment (vpj) [inline app/ui/src/network.ts:220]: Shouldn't this be NetworkError?

File: app/ui/src/components/user_messages.ts [hunk 2]
```diff
@@ -36,16 +37,13 @@ export class UserMessages {
...
-                + ((error.message != null) ? " " + error.message : "")
-        } else if (error instanceof Error) {
-            description = error.message
+            description += error.toString()
+        } else {
+            description += errrToString(error)
         }
-        this.error(message + (description ? `: ${description}` : '') + (error instanceof NetworkError &&
-        error?.stackTrace ? `\n${error.stackTrace}` : ''))
+        this.error(description)
```
Comment (vpj) [inline app/ui/src/components/user_messages.ts:46]: In `error` we can end the message with eclipses and have a more link to open the full error content on a new tab `<a href="data:text/html,<pre>full error</pre>">more</a>`

Comment (lakshith-403) [reply app/ui/src/components/user_messages.ts:67]: 7be27cea9539d5b1cc3e21df50b69b07b122a4d7

File: app/ui/src/network.ts (hunk 1 above)
Comment (lakshith-403) [reply app/ui/src/network.ts]: 58bf3c26f5e6a0c12f4d65d7de40629a3a5a40aa

**Pull request merged**